management/management_request.go

```diff
@@ -84,27 +84,35 @@ func (m *Management) Do(req *http.Request) (*http.Response, error) {
 }
 
 // Request combines NewRequest and Do, while also handling decoding of response payload.
-func (m *Management) Request(method, uri string, v interface{}, options ...RequestOption) error {
-	request, err := m.NewRequest(method, uri, v, options...)
+func (m *Management) Request(method, uri string, payload interface{}, options ...RequestOption) error {
+	request, err := m.NewRequest(method, uri, payload, options...)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to create a new request: %w", err)
 	}
 
 	response, err := m.Do(request)
 	if err != nil {
-		return fmt.Errorf("request failed: %w", err)
+		return fmt.Errorf("failed to send the request: %w", err)
 	}
+	defer response.Body.Close()
 
-	if response.StatusCode < http.StatusOK || response.StatusCode >= http.StatusBadRequest {
-		return newError(response.Body)
+	// If the response contains a client or a server error then return the error.
+	if response.StatusCode >= http.StatusBadRequest {
+		return newError(response)
 	}
 
-	if response.StatusCode != http.StatusNoContent && response.StatusCode != http.StatusAccepted {
```

Comment:
This check was replaced by `if len(responseBody) > 0 && string(responseBody) != "{}"`